Appc 1051 behaviour after install wallet

brunovarelaAptoide opened the pull request:
**What does this PR do?**

Reconnect with Wallet AIDL after wallet install.

- Added a wait() and notify routines in StubHelper.

-Toolbox change to invoke Billing functions in new threads

-Wallet utils changed to invoke the popup in ui main thread.

**Database changed?**

 No

**Where should the reviewer start?**
-AppcoinsBillingStubHelper
-WalletUtils

**How should this be manually tested?**

 - Change the package name to trivialdrivesample.test to test the inapps.

**What are the relevant tickets?**

1051

**Questions:**

   Does this add new dependencies which need to be added? (Eg. new libs, new keys, etc.) 
No



**Code Review Checklist**

- [ ] Architecture
- [ ] Documentation on public interfaces
- [ ] Database changed?
- [ ] If yes - Database migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Functional QA tests pass

brunovarelaAptoide committed:
Changes

- Removed unusued argument

brunovarelaAptoide committed:
Reconnect to AIDL after wallet install

- Added a wait() and notify routines in StubHelper.
-Toolbox change to invoke Billing functions in new threads
-Wallet utils changed to invoke the popup in ui main thread.

nunomonteiro commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/AppCoinsAndroidBillingRepository.java:
```diff
@@ -54,25 +53,29 @@ public AppCoinsAndroidBillingRepository(int apiVersion, String packageName, Cont
     } catch (RemoteException e) {
-      throw new ServiceConnectionException(e.getMessage());
+      e.printStackTrace();
+      throw new ServiceConnectionException();
```
Why did you take the e.getMessage() out of the exception?

nunomonteiro commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/AppcoinsBillingStubHelper.java:
```diff
@@ -2,66 +2,201 @@
+          synchronized (lockThread) {
+            lockThread.wait();
+          }
+        }
```
Isn't this missing the catch block?

catch (RemoteException e) {
        e.printStackTrace();
        responseWs.putInt(Utils.RESPONSE_CODE, ResponseCode.SERVICE_UNAVAILABLE.getValue());
      } catch (InterruptedException e) {
        e.printStackTrace();
        responseWs.putInt(Utils.RESPONSE_CODE, ResponseCode.SERVICE_UNAVAILABLE.getValue());
      }

nunomonteiro reviewed (changes requested):
Left inline comment

brunovarelaAptoide committed:
PR Changes

- Changes Exception
- To keep coerence in WalletUtils remove RunOnUIThread, and placed it in Stub

Trinkes commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/AppcoinsBillingStubHelper.java:
```diff
@@ -1,67 +1,215 @@
 
+  private Object lockThread;
+  private static AppcoinsBilling serviceAppcoinsBilling;
+  private boolean wasServiceBinded = false;
```
can you rename this variable to "isServiceBound"?

Trinkes commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/AppcoinsBillingStubHelper.java:
```diff
@@ -1,67 +1,215 @@
+        }
+
+        return serviceAppcoinsBilling.isBillingSupported(apiVersion, packageName, type);
+      } catch (RemoteException e) {
```
you can colapse the catch clauses

Trinkes commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/AppcoinsBillingStubHelper.java:
```diff
@@ -1,67 +1,215 @@
-    return response;
+    if (WalletUtils.hasWalletInstalled()) {
+      try {
+        if (!wasServiceBinded) {
```
every time you access to wasServiceBinded variable, you should synchronise it so you don't run createRepository method multiple times

Trinkes commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/AppcoinsBillingStubHelper.java:
```diff
@@ -1,67 +1,215 @@
+    final AppCoinsAndroidBillingRepository repository =
+        new AppCoinsAndroidBillingRepository(Utils.API_VERSION, WalletUtils.walletPackageName);
+
+    RepositoryServiceConnection connection =
```
if we are including RepositoryServiceConnection class  here, it can't implement RepositoryConnection because there is no reason to add this interface to migrator's generated code

brunovarelaAptoide committed:
Changes Pull Request

- exception handling
- stop using a repository connection
- sycronized the whole block with isServiceBlock
- Changed variable to isServiceBlock
- used the sycronized tag in getPurchases and consumePurchase

Trinkes committed:
rename GetSkuDetailsService method to getSkuDetailsService

Trinkes committed:
remove synchronization from Sub Helper methods

Trinkes committed:
Merge branch 'APPC-1051-Develop-Billing-standalone-SDK' into APPC-1051-Behaviour-after-install-wallet

# Conflicts:
#	android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/AppcoinsBillingStubHelper.java
#	android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/CatapultBillingAppCoinsFactory.java
